diff --git a/meson.build b/meson.build
index 27d78ab1..84f991e7 100644
--- a/meson.build
+++ b/meson.build
@@ -75,6 +75,7 @@ if not headers_only
 	add_project_arguments('-nostdinc', '-fno-builtin', '-ffreestanding', language: ['c', 'cpp'])
 	add_project_arguments('-Werror=misleading-indentation', language: ['c', 'cpp'])
 	add_project_arguments('-fno-rtti', '-fno-exceptions', language: 'cpp')
+	add_project_arguments('-mno-avx', '-mno-avx2', '-mno-avx512f', language: ['c', 'cpp'])
 	add_project_link_arguments('-nostdlib', language: ['c', 'cpp'])
 
 	if get_option('buildtype').startswith('debug')
@@ -274,6 +275,10 @@ elif host_machine.system() == 'nyaux'
 	rtld_include_dirs += include_directories('sysdeps/nyaux/include')
 	libc_include_dirs += include_directories('sysdeps/nyaux/include')
 	subdir('sysdeps/nyaux')
+elif host_machine.system() == 'modulos'
+  rtld_include_dirs += include_directories('sysdeps/modulos/include')
+  libc_include_dirs += include_directories('sysdeps/modulos/include')
+  subdir('sysdeps/modulos')
 else
 	error('No sysdeps defined for OS: ' + host_machine.system())
 endif
diff --git a/options/internal/include/mlibc/internal-sysdeps.hpp b/options/internal/include/mlibc/internal-sysdeps.hpp
index 8c058cfb..6e819043 100644
--- a/options/internal/include/mlibc/internal-sysdeps.hpp
+++ b/options/internal/include/mlibc/internal-sysdeps.hpp
@@ -7,6 +7,7 @@
 #include <abi-bits/vm-flags.h>
 #include <bits/off_t.h>
 #include <bits/ssize_t.h>
+#include <bits/ansi/timespec.h>
 #include <abi-bits/stat.h>
 #include <mlibc/fsfd_target.hpp>
 
